Packs/QRadar/Integrations/QRadar_v3/QRadar_v3.py

```diff
@@ -787,51 +794,63 @@ def insert_values_to_reference_set_polling(client: Client,
+                element_type = args.get('element_type', '')
+                timeout_type = args.get('timeout_type')
+                time_to_live = args.get('time_to_live')
+                try:
+                    client.reference_sets_list(ref_name=ref_name)
+                except DemistoException as e:
+                    # Create reference set if does not exist
+                    if e.message and f'{ref_name} does not exist' in e.message:
+                        # if this call fails, raise an error and stop command execution
+                        client.reference_set_create(ref_name, element_type, timeout_type, time_to_live)
+                    else:
+                        raise e
+
+                search_indicators = IndicatorsSearcher(page=page)
+                indicators = search_indicators.search_indicators_by_version(query=query, size=limit).get('iocs', [])
+                indicators_data = [{'Indicator Value': indicator.get('value'), 'Indicator Type': indicator.get('indicator_type')}
+                                   for indicator in indicators if 'value' in indicator and 'indicator_type' in indicator]
+                values = [indicator.get('Indicator Value', '') for indicator in indicators_data]
+                if not indicators_data:
+                    return PollResult(CommandResults(
+                        readable_output=f'No indicators were found for reference set {ref_name}',
+                    ))
+
+            if not values:
+                raise DemistoException('Value to insert must be given.')
+            source = args.get('source')
+            date_value = argToBoolean(args.get('date_value', False))
+            fields = args.get('fields')
+
+            if date_value:
+                values = [get_time_parameter(value, epoch_format=True) for value in values]
+            if use_old_api:
+                response = client.reference_set_bulk_load(ref_name, values, fields)
+            else:
+                response = client.reference_set_entries(ref_name, values, fields, source, timeout=300)
+                args['task_id'] = response.get('id')
+        if not use_old_api:
+            response = client.get_reference_data_bulk_task_status(args["task_id"])
+    except (DemistoException, requests.Timeout) as e:
+        if 'task_id' in args:
+            print_debug_msg(
+                f"Polling task status {args['task_id']} failed due to {e}. "
+                f"Will try to poll task status again in the next interval."
+            )
         else:
-            response = client.reference_set_entries(ref_name, values, fields, source)
-            args['task_id'] = response.get('id')
-    if not use_old_api:
-        response = client.get_reference_data_bulk_task_status(args["task_id"])
+            print_debug_msg(
+                f"Failed inserting values to reference due to {e}, will retry in the insert in the next interval"
+            )
+        response = {}
```

Comment:
if the timeout won't be enough in some situations, and we will have to keep polling over and over again.

Reply:
yeah, however polling commands have timeout (by default 600 seconds), so the command would exit anyway